tests/src/utils/function.test.ts

```diff
@@ -18,5 +18,97 @@ describe( 'function utils', () => {
 
 			expect( fn ).toHaveBeenCalledTimes( 3 );
 		} );
+
+		it( 'should cache repeated calls of async function', async () => {
+			const fn = vi.fn().mockImplementation( async ( arg: number ): Promise< string > => {
+				await Promise.resolve();
+				return `called with: ${ arg }`;
+			} );
+			const memoized = memoizeFn( fn );
+
+			expect( await memoized( 1 ) ).toEqual( 'called with: 1' );
+			expect( await memoized( 1 ) ).toEqual( 'called with: 1' );
+			expect( await memoized( 2 ) ).toEqual( 'called with: 2' );
+			expect( await memoized( 2 ) ).toEqual( 'called with: 2' );
+			expect( await memoized( 2 ) ).toEqual( 'called with: 2' );
+			expect( await memoized( 3 ) ).toEqual( 'called with: 3' );
+			expect( await memoized( 3 ) ).toEqual( 'called with: 3' );
+
+			expect( fn ).toHaveBeenCalledTimes( 3 );
+		} );
+
+		it( 'should cache a then-able promise if that is returned by the function', async () => {
+			const fn = vi.fn().mockImplementation( async ( arg: number ): Promise< string > => {
+				await Promise.resolve();
+				return `called with: ${ arg }`;
+			} );
+			const memoized = memoizeFn< ( arg: number ) => Promise< string > >( fn );
+
+			expect( await memoized( 1 ) ).toEqual( 'called with: 1' );
+			expect( await memoized( 1 ) ).toEqual( 'called with: 1' );
+
+			const thenPromise = new Promise< string >( ( resolve, reject ) => {
+				memoized( 1 ).then( resolve ).catch( reject );
+			} );
+
+			expect( await thenPromise ).toEqual( 'called with: 1' );
```

Comment:
Does this provide useful extra assurance over using `await memoized( 1 )` twice?

Reply:
Yes, if the original function returns a promise, it ensures that the value returned by the memoized function is also a promise (as opposed to the awaited value). This was a bug in my initial commit.

Reply:
Oh, I see. That makes sense now. 👍